imagenet/main.py

```diff
@@ -200,16 +232,29 @@ def train(train_loader, model, criterion, optimizer, epoch):
         top1.update(prec1[0], input.size(0))
         top5.update(prec5[0], input.size(0))
 
+        loss = loss*args.loss_scale
         # compute gradient and do SGD step
-        optimizer.zero_grad()
-        loss.backward()
-        optimizer.step()
+
+        if args.fp16:
+            model.zero_grad()
+            loss.backward()
+            if args.loss_scale != 1:
```

Comment:
Loss scaling is to help maintain a well behaved range during back propagation, preventing gradients from getting truncated to 0. Running `(loss / args.loss_scale).backward()` would be the same as not having loss scaling.